File: 1249-minimum-remove-to-make-valid-parentheses/1249-minimum-remove-to-make-valid-parentheses.cpp
class Solution {
public:
    string minRemoveToMakeValid(string s) {
        int i = 0,count = 0;
        string result;
        stack<pair<char,int>> st;
        while(i<s.size())
        {
            if(s[i]=='(')
            {
                result+=s[i];
                int ind = result.size()-1;
                st.push({s[i],ind});
            }
            else if(s[i]==')')
            {
                if(!(st.empty()))
                {
                    result+=s[i];
                    st.pop();
                }
            }
            else
            {
                result+=s[i];
            }
            i++;
        }
        cout<<result;
        int j = 0;
        while(st.empty()==false)
        {
            int ind = st.top().second;
            result.erase(result.begin()+ind);
            st.pop();
        }
        return result;
    }
};

File: 1249-minimum-remove-to-make-valid-parentheses/1249-minimum-remove-to-make-valid-parentheses.py
class Solution:
    def minRemoveToMakeValid(self, s: str) -> str:
        stack = []
        for i,char in enumerate(s):
            if not stack and char in {'(',')'}:
                stack.append((i,char))

            elif char==')' and stack[-1][1]=='(':
                stack.pop()
            elif char in {'(',')'}:
                stack.append((i,char))

        ans = ""
        sett = set()
        for i,j in stack:
            sett.add(i)

        for i,c in enumerate(s):
            if i not in sett:
                ans+=c
        return ans

        return ""


            

